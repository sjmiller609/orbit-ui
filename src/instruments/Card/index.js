'use strict'
import React from 'react'
import PropTypes from 'prop-types'
import classnames from 'classnames'
import s from './styles.scss'

const Card = ({ children, footer, header, className }) => {
  return (
    <div className={classnames(s.card, className)}>
      {header && <header>{header}</header>}
      <main>{children}</main>
      {footer && <footer>{footer}</footer>}
    </div>
  )
}

Card.propTypes = {
  header: PropTypes.element,
  children: PropTypes.element.isRequired,
  footer: PropTypes.element,
  className: PropTypes.string,
}

export default Card
