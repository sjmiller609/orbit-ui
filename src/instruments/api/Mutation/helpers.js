const has = (str, find) => ~str.toLowerCase().indexOf(find.toLowerCase())

export const formErrors = ({ error, errors }) => {
  if (!error) return

  let err = JSON.stringify(error).toLowerCase()
  return errors.find(e => {
    if (e.key && has(err, e.key)) return e
    else if (e.allKeys && e.allKeys.every(k => has(err, k))) return e
    else if (e.someKeys && e.someKeys.some(k => has(err, k))) return e
  })
}
