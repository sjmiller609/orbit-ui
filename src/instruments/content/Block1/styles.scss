@import 'src/styles/mixins.scss';

$width: 54rem;

.content {
  margin: auto;
  width: 100%;
  max-width: $width;
  padding: 1.5rem 0 3rem;

  @include spacing;
}

.left {
  width: $width * 0.3;
  flex-grow: 1;
}

.right {
  width: $width * 0.6;
  flex-grow: 1;
}

.left,
.right {
  margin: 1rem;
}
