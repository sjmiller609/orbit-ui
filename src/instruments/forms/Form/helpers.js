// faltten and stringify nested keys
export const unpack = (obj, ns, hash) => {
  if (!obj) return
  return Object.entries(obj).reduce((acc, [key, value]) => {
    const keypath = ns ? `${ns}.${key}` : key
    if (value && typeof value === 'object') {
      return unpack(value, keypath, acc)
    }
    acc[keypath] = value
    return acc
  }, hash || {})
}

const roll = (keys, obj, value) => {
  const key = keys[0]
  // case 1 - not obj
  if (keys.length === 1) {
    obj[key] = value
    return obj
  }
  // case 2 - doesn't exist
  if (!obj[key]) {
    obj[key] = isNaN(keys[1]) ? {} : []
  }
  // case 3 - exists
  obj[key] = roll(
    keys.slice(1),
    Array.isArray(obj[key]) ? [...obj[key]] : { ...obj[key] },
    value
  )

  return obj
}

// convert back into obj
export const pack = (obj, type = {}) => {
  return Object.entries(obj).reduce((acc, [key, value]) => {
    const keys = key.split('.')

    return roll(keys, acc, value)
  }, type)
}

export const packChild = ({ name, obj }) => {
  const obj2 = {}
  let isArray
  Object.entries(obj).forEach(k => {
    if (name.length < k[0].length && k[0].indexOf(name) === 0) {
      const n = k[0].slice(name.length + 1)
      // must be true for all matching
      isArray = !isNaN(n.slice(0, n.indexOf('.')))
      obj2[n] = k[1]
    }
  })
  if (Object.keys(obj2).length) return pack(obj2, isArray ? [] : {})
  return null
}

export const removeChild = ({ name, obj = {} }) => {
  const obj2 = {}
  Object.entries(obj).forEach(k => {
    if (name.length >= k[0].length || k[0].indexOf(name) !== 0) {
      obj2[k[0]] = k[1]
    }
  })
  return obj2
}
