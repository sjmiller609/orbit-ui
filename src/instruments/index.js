// NOTE: Any instruments importing other instruments must come AFTER

export { default as Row } from './flex/Row'
export { default as Box } from './flex/Box'

export { default as Card } from './cards/Card'
export { default as Table } from './cards/Table'
export { default as TableRow } from './cards/Table/TableRow'
export { default as CardForm } from './cards/CardForm'

export { default as Module } from './layouts/Module'
export { default as Header } from './layouts/Header'

export { default as Loading } from './animations/Loading'
export { default as LoadingDots } from './animations/LoadingDots'

export { default as Search } from './forms/Search'
export { default as Form } from './forms/Form'
export { default as TextField } from './forms/TextField'
export { default as Password } from './forms/Password'
export { default as Checkbox } from './forms/Checkbox'

export { default as Link } from './links/Link'
export { default as Button } from './links/Button'

export { default as Icon } from './icons/Icon'
export { default as Logo } from './icons/Logo'

export { default as P } from './text/P'
export { default as Mini } from './text/Mini'
export { default as ShowDate } from './text/ShowDate'
export { default as B } from './text/B'

export { default as H1 } from './text/H1'
export { default as H2 } from './text/H2'
export { default as H3 } from './text/H3'
export { default as H4 } from './text/H4'
export { default as H5 } from './text/H5'

export { default as Menu } from './menus/Menu'
export { default as Level1 } from './menus/Level1'

export { default as Snackbar } from './ui/Snackbar'

export { default as ContextUI } from './context/UI/Get'
export { default as ProviderUI } from './context/UI'

export { default as Query } from './api/Query'
export { default as Mutation } from './api/Mutation'
