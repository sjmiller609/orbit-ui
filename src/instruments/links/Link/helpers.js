// check if url is external
export const externalUrl = url => {
  if (typeof url === 'object') return false
  const path = /^https?:\/\/|^\/\//i
  if (path.test(url) && !~url.indexOf(window.location.origin)) return true
  return
}
