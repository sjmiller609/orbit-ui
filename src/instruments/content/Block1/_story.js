import React from 'react'
import { storiesOf } from '@storybook/react'
import { withKnobs, text } from '@storybook/addon-knobs'
import { Block1, P } from 'instruments'

storiesOf('Instruments|Content.Block1', module)
  .addDecorator(withKnobs)
  .add('No props', () => <Block1 />)
  .add('w/ props', () => (
    <Block1
      title={text('Title', 'Mock Title')}
      text={text('Text', 'String of mock text')}>
      <P>
        {text('Child Text', 'String of mock text located in the child prop')}
      </P>
    </Block1>
  ))
  .add('w/ left prop', () => (
    <Block1
      left={<P>Left column</P>}
      title={text('Title', 'Mock Title')}
      text={text('Text', 'String of mock text')}>
      <P>
        {text('Child Text', 'String of mock text located in the child prop')}
      </P>
    </Block1>
  ))
  .add('w/ right prop', () => (
    <Block1
      right={<P>Right column</P>}
      title={text('Title', 'Mock Title')}
      text={text('Text', 'String of mock text')}>
      <P>
        {text('Child Text', 'String of mock text located in the child prop')}
      </P>
    </Block1>
  ))
