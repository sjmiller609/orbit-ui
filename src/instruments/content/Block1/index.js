'use strict'
import React from 'react'
import PropTypes from 'prop-types'
import classnames from 'classnames'
import s from './styles.scss'
import { Row, Box, H1, P } from 'instruments'

const Block1 = ({ left, right, title, text, children, className }) => {
  return (
    <Row className={classnames(s.content, className)} wrap>
      {left && <Box className={s.left}>{left}</Box>}
      <Box className={s.right} align="flex-start">
        {right || (
          <React.Fragment>
            {title && <H1>{title}</H1>}
            {text && <P>{text}</P>}
            {children}
          </React.Fragment>
        )}
      </Box>
    </Row>
  )
}

Block1.propTypes = {
  left: PropTypes.element,
  right: PropTypes.element,
  title: PropTypes.string,
  text: PropTypes.string,
  children: PropTypes.oneOfType([PropTypes.array, PropTypes.element]),
  className: PropTypes.string,
}

export default Block1
