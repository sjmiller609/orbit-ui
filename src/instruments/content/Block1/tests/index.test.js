import React from 'react'
import { mountWrap } from 'helpers/tests'
import { Link } from 'instruments'
import Block1 from '../index'

const props = {
  left: <Link>Test</Link>,
  right: <Link>Test</Link>,
  title: 'Test',
  text: 'This is a test',
  children: ['<p>This is a test</p>'],
  className: 'testClassString',
}

describe('Block1', () => {
  it('should be defined', () => {
    expect(Block1).toBeDefined()
  })

  it('renders correctly', () => {
    const wrapper = mountWrap(<Block1 {...props} />)
    expect(wrapper).toMatchSnapshot()
  })

  it('handles no right', () => {
    const wrapper = mountWrap(<Block1 {...props} right={null} />)
    expect(wrapper).toMatchSnapshot()
  })

  it('handles no left', () => {
    const wrapper = mountWrap(<Block1 {...props} left={null} />)
    expect(wrapper).toMatchSnapshot()
  })

  it('handles no text', () => {
    const wrapper = mountWrap(<Block1 {...props} text={null} />)
    expect(wrapper).toMatchSnapshot()
  })

  it('handles no title', () => {
    const wrapper = mountWrap(<Block1 {...props} title={null} />)
    expect(wrapper).toMatchSnapshot()
  })
})
