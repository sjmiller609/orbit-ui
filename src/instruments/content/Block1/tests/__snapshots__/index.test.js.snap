// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Block1 handles no left 1`] = `
<Block1
  className="testClassString"
  left={null}
  right={
    <Link>
      Test
    </Link>
  }
  text="This is a test"
  title="Test"
>
  <Row
    className="content testClassString"
    wrap={true}
  >
    <Flex
      className="content testClassString"
      flow="row"
      wrap={true}
    >
      <div
        className="flex row undefined wrap content testClassString"
      >
        <Box
          align="flex-start"
          className="right"
        >
          <Flex
            align="flex-start"
            className="right"
            flow="box"
          >
            <div
              className="flex box undefined align-flex-start right"
            >
              <Link>
                <A
                  className="link"
                >
                  <a
                    className="link"
                  >
                    Test
                  </a>
                </A>
              </Link>
            </div>
          </Flex>
        </Box>
      </div>
    </Flex>
  </Row>
</Block1>
`;

exports[`Block1 handles no right 1`] = `
<Block1
  className="testClassString"
  left={
    <Link>
      Test
    </Link>
  }
  right={null}
  text="This is a test"
  title="Test"
>
  <Row
    className="content testClassString"
    wrap={true}
  >
    <Flex
      className="content testClassString"
      flow="row"
      wrap={true}
    >
      <div
        className="flex row undefined wrap content testClassString"
      >
        <Box
          className="left"
        >
          <Flex
            className="left"
            flow="box"
          >
            <div
              className="flex box undefined left"
            >
              <Link>
                <A
                  className="link"
                >
                  <a
                    className="link"
                  >
                    Test
                  </a>
                </A>
              </Link>
            </div>
          </Flex>
        </Box>
        <Box
          align="flex-start"
          className="right"
        >
          <Flex
            align="flex-start"
            className="right"
            flow="box"
          >
            <div
              className="flex box undefined align-flex-start right"
            >
              <H1>
                <h1
                  className="h1"
                >
                  Test
                </h1>
              </H1>
              <P>
                <p
                  className="p"
                >
                  This is a test
                </p>
              </P>
              &lt;p&gt;This is a test&lt;/p&gt;
            </div>
          </Flex>
        </Box>
      </div>
    </Flex>
  </Row>
</Block1>
`;

exports[`Block1 handles no text 1`] = `
<Block1
  className="testClassString"
  left={
    <Link>
      Test
    </Link>
  }
  right={
    <Link>
      Test
    </Link>
  }
  text={null}
  title="Test"
>
  <Row
    className="content testClassString"
    wrap={true}
  >
    <Flex
      className="content testClassString"
      flow="row"
      wrap={true}
    >
      <div
        className="flex row undefined wrap content testClassString"
      >
        <Box
          className="left"
        >
          <Flex
            className="left"
            flow="box"
          >
            <div
              className="flex box undefined left"
            >
              <Link>
                <A
                  className="link"
                >
                  <a
                    className="link"
                  >
                    Test
                  </a>
                </A>
              </Link>
            </div>
          </Flex>
        </Box>
        <Box
          align="flex-start"
          className="right"
        >
          <Flex
            align="flex-start"
            className="right"
            flow="box"
          >
            <div
              className="flex box undefined align-flex-start right"
            >
              <Link>
                <A
                  className="link"
                >
                  <a
                    className="link"
                  >
                    Test
                  </a>
                </A>
              </Link>
            </div>
          </Flex>
        </Box>
      </div>
    </Flex>
  </Row>
</Block1>
`;

exports[`Block1 handles no title 1`] = `
<Block1
  className="testClassString"
  left={
    <Link>
      Test
    </Link>
  }
  right={
    <Link>
      Test
    </Link>
  }
  text="This is a test"
  title={null}
>
  <Row
    className="content testClassString"
    wrap={true}
  >
    <Flex
      className="content testClassString"
      flow="row"
      wrap={true}
    >
      <div
        className="flex row undefined wrap content testClassString"
      >
        <Box
          className="left"
        >
          <Flex
            className="left"
            flow="box"
          >
            <div
              className="flex box undefined left"
            >
              <Link>
                <A
                  className="link"
                >
                  <a
                    className="link"
                  >
                    Test
                  </a>
                </A>
              </Link>
            </div>
          </Flex>
        </Box>
        <Box
          align="flex-start"
          className="right"
        >
          <Flex
            align="flex-start"
            className="right"
            flow="box"
          >
            <div
              className="flex box undefined align-flex-start right"
            >
              <Link>
                <A
                  className="link"
                >
                  <a
                    className="link"
                  >
                    Test
                  </a>
                </A>
              </Link>
            </div>
          </Flex>
        </Box>
      </div>
    </Flex>
  </Row>
</Block1>
`;

exports[`Block1 renders correctly 1`] = `
<Block1
  className="testClassString"
  left={
    <Link>
      Test
    </Link>
  }
  right={
    <Link>
      Test
    </Link>
  }
  text="This is a test"
  title="Test"
>
  <Row
    className="content testClassString"
    wrap={true}
  >
    <Flex
      className="content testClassString"
      flow="row"
      wrap={true}
    >
      <div
        className="flex row undefined wrap content testClassString"
      >
        <Box
          className="left"
        >
          <Flex
            className="left"
            flow="box"
          >
            <div
              className="flex box undefined left"
            >
              <Link>
                <A
                  className="link"
                >
                  <a
                    className="link"
                  >
                    Test
                  </a>
                </A>
              </Link>
            </div>
          </Flex>
        </Box>
        <Box
          align="flex-start"
          className="right"
        >
          <Flex
            align="flex-start"
            className="right"
            flow="box"
          >
            <div
              className="flex box undefined align-flex-start right"
            >
              <Link>
                <A
                  className="link"
                >
                  <a
                    className="link"
                  >
                    Test
                  </a>
                </A>
              </Link>
            </div>
          </Flex>
        </Box>
      </div>
    </Flex>
  </Row>
</Block1>
`;
